Cannot find name 'URLSearchParams'

#### What would you like to discuss?

I've just upgraded to `got@10` and got this TypeScript error:

```
tsc
node_modules/got/dist/source/utils/options-to-url.d.ts:12:71 - error TS2304: Cannot find name 'URLSearchParams'.
```

I figured it's because in my `tsconfig.json` `libs` I don't have `dom`. I'm consciously not using `dom` there to avoid importing/using Browser types that I never need in Node.js project (it caused me issues in the past, e.g global dom `Request` type clashes with Express.js `Request` type).

Is there a way to use `got@10` in a TypeScript project without `dom` in `libs`?

#### Checklist

- [x] I have read the documentation.

> #### What would you like to discuss?
> 
> I've just upgraded to `got@10` and got this TypeScript error:
> 
> ```
> tsc
> node_modules/got/dist/source/utils/options-to-url.d.ts:12:71 - error TS2304: Cannot find name 'URLSearchParams'.
> ```
> 
> I figured it's because in my `tsconfig.json` `libs` I don't have `dom`. I'm consciously not using `dom` there to avoid importing/using Browser types that I never need in Node.js project (it caused me issues in the past, e.g global dom `Request` type clashes with Express.js `Request` type).
> 
> Is there a way to use `got@10` in a TypeScript project without `dom` in `libs`?
> #### Checklist
> 
>     * [x]  I have read the documentation.

Is it possible to share some reproduce-able code? I already know the culprit, but my local setup miraculously still compiles.

@sindresorhus This is actually a regression of https://github.com/sindresorhus/got/issues/876#issuecomment-558813981. Apparently the issue is much deeper - because even though it would compile for Got's project, the augmented global types won't get populated to the users and thus TypeScript is doing its thing to try resolving the types. I guess we have to resort to moving `url.d.ts` into our types file so that it will get bundled, or resort to using the imported version from `url` until the types are properly fixed from DefinitelyTyped.

@szmarczak Thoughts?

--- a/source/utils/types.ts
+++ b/source/utils/types.ts
@@ -247,3 +247,44 @@ export interface CancelableRequest<T extends Response | Response['body']> extend
 	buffer(): CancelableRequest<Buffer>;
 	text(): CancelableRequest<string>;
 }
+
+// TODO: Remove this when https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34960 is resolved
+declare global {
+	// @ts-ignore TypeScript complains about duplicated identifier.
+	class URL {
+		readonly origin: string;
+		readonly searchParams: URLSearchParams;
+		hash: string;
+		host: string;
+		hostname: string;
+		href: string;
+		password: string;
+		pathname: string;
+		port: string;
+		protocol: string;
+		search: string;
+		username: string;
+
+		constructor(input: string, base?: string | URL);
+		toString(): string;
+		toJSON(): string;
+	}
+
+	// @ts-ignore TypeScript complains about duplicated identifier.
+	class URLSearchParams implements Iterable<[string, string]> {
+		constructor(init?: URLSearchParams | string | {[key: string]: string | string[] | undefined} | Iterable<[string, string]> | Array<[string, string]>);
+		append(name: string, value: string): void;
+		delete(name: string): void;
+		entries(): IterableIterator<[string, string]>;
+		forEach(callback: (value: string, name: string, searchParams: this) => void): void;
+		get(name: string): string | null;
+		getAll(name: string): string[];
+		has(name: string): boolean;
+		keys(): IterableIterator<string>;
+		set(name: string, value: string): void;
+		sort(): void;
+		toString(): string;
+		values(): IterableIterator<string>;
+		[Symbol.iterator](): IterableIterator<[string, string]>;
+	}
+}

